plugin/src/main/java/org/opensearch/ml/model/MLModelAndNodeManager.java

```diff
@@ -0,0 +1,294 @@
+                throw new RuntimeException(e);
+            }
+        }
+
+        // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+        saveLatestReTryTimes(nodeId, reTryTimes);
+    }
+
+    /**
+     *
+     * @param nodeId
+     * @param modelId
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeAndModelId(String nodeId, String modelId) {
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, new String[] { nodeId }, false, false);
+        LoadModelResponse loadModelResponse = client.execute(MLLoadModelAction.INSTANCE, mlLoadModelRequest).actionGet();
+
+        if (loadModelResponse == null) {
+            throw new RuntimeException("Can't auto reload model " + modelId + " by node " + nodeId);
+        }
+    }
+
+    @VisibleForTesting
+    Integer getReTryTimes(String nodeId) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.fetchSource(new String[] { MODEL_LOAD_RETRY_TIMES_FIELD }, null);
+        QueryBuilder queryBuilder = new TermQueryBuilder(NODE_ID_FIELD, nodeId);
+        searchSourceBuilder.query(queryBuilder);
+        SearchRequest searchRequest = new SearchRequest().source(searchSourceBuilder).indices(ML_MODEL_RELOAD_INDEX);
+        SearchResponse response = client.execute(MLModelSearchAction.INSTANCE, searchRequest).actionGet(5000);
+
+        if (response == null) {
+            throw new RuntimeException("can't get retry times by " + nodeId);
+        }
+        SearchHits searchHits = response.getHits();
+        if (searchHits == null) {
+            throw new RuntimeException("can't get retry times by " + nodeId);
+        }
+        SearchHit[] hits = searchHits.getHits();
+        if (CollectionUtils.isEmpty(hits)) {
+            throw new RuntimeException("can't get retry times by " + nodeId);
+        }
+
+        for (SearchHit hit : hits) {
+            Map<String, Object> sourceAsMap = hit.getSourceAsMap();
+            return (Integer) sourceAsMap.get(MODEL_LOAD_RETRY_TIMES_FIELD);
+        }
+
+        throw new RuntimeException("can't get retry times by " + nodeId);
+    }
+
+    @VisibleForTesting
+    boolean isExistedIndex(String indexName) {
+        IndicesExistsRequest existsRequest = new IndicesExistsRequest(indexName);
+        IndicesExistsResponse exists = client.execute(IndicesExistsAction.INSTANCE, existsRequest).actionGet(5000);
+
+        return exists.isExists();
+    }
+
+    @VisibleForTesting
+    void saveLatestReTryTimes(String nodeId, Integer reTryTimes) {
+        Map<String, Object> content = new HashMap<>();
+        content.put(NODE_ID_FIELD, nodeId);
+        content.put(MODEL_LOAD_RETRY_TIMES_FIELD, reTryTimes);
+
+        if (isExistedIndex(ML_MODEL_RELOAD_INDEX)) {
+            // update data under the index
+            UpdateRequest updateRequest = new UpdateRequest();
```

Comment:
We can try always using IndexRequest with docId (if exist), this is a upsert operation.